__tests__/conflict-test.ts

```diff
@@ -0,0 +1,37 @@
+describe('merge conflict resolution', () => {
```

Comment:
Let's unnest all these describes. We don't need this level of nesting in our tests. I think you can simply have the top level `describe` and nested `it` calls inside.